routes/documents.js

```diff
@@ -24,8 +24,13 @@ router.get('/:docId', (req, res, next) => {
     .catch(next);
 });
 
-router.post('/', validator(DocumentBodySchema), (req, res, next) => {
-  Document.create(req.body)
+router.post('/', validator(DocumentBodySchema), async (req, res, next) => {
+  const id = await Document.create(req.body)
+    .then(document => document.id)
+    .catch(next);
+  Document.findById(id)
+    .populate('owner', '-createdAt -updatedAt')
```

Comment:
Check if you're not exposing users' passwords here.

Reply:
Exposing of user password blocked in /models/user.js:
UserSchema:
...
  email: {
    type: String,
    required: true,
    unique: true,
    ***select: false***,
  },
  password: {
    type: String,
    required: true,
    ***select: false***,
  },
....